### PR TITLE
Update patterns.rst

### PR DESCRIPTION
**Pull request by JonasAls89**

### CODE REVIEW

File: patterns.rst [hunk 1]
```diff
@@ -101,6 +101,14 @@ Optimistic locking
 ------------------
 Should be added via an external transform and then two hash values should be compared. In case of difference, discard entity.
 
+Subsets
```
Comment (BaardBouvet) [inline patterns.rst:104]: I think we should call it "Only process the relevant entities from a global using subset"

File: patterns.rst [hunk 2]
```diff
@@ -101,6 +101,14 @@ Optimistic locking
...
+-------
+If your source supports it, you should use the :ref:`subset <dataset_source>` property to pick subsets of a complete dataset, i.e: after a merge source has been used upstream.
```
Comment (BaardBouvet) [inline patterns.rst:106]: The global will always support subsets, so this is irrelevant.

**Review by BaardBouvet — CHANGES_REQUESTED** [review]

**Comment by JonasAls89** [comment]
Trying to sum up on the 50 plus threads about this. Let me know if something should be restated.

File: patterns.rst [hunk 3]
```diff
@@ -101,6 +101,14 @@ Optimistic locking
 ------------------
 Should be added via an external transform and then two hash values should be compared. In case of difference, discard entity.
 
+Only process the relevant entities from a global using subset
+-------------------------------------------------------------
+You should use the :ref:`subset <dataset_source>` property to pick subsets of a complete dataset, i.e: after a merge source has been used upstream. 
+
+.. note::
+
+  This pattern is sensitive to multiple versions of the entity id in one complete pipe run, i.e: if entities move between subsets, then you need to deal with these moves, as they will create old entity versions, which will be deletes.
```
Comment (grove) [inline patterns.rst:110]: I don’ think we should call it move between subsets.  An entity can be part of many subsets at the same time. It is more about an entity being added to a subset and then later removed from it. Just the same was as an entity is added to dataset and then later removed from it.

File: patterns.rst [hunk 4]
```diff
@@ -101,6 +101,12 @@ Optimistic locking
 ------------------
 Should be added via an external transform and then two hash values should be compared. In case of difference, discard entity.
 
+A “Best Effort” to extract entities using create and subset
+-----------------------------------------------------------
+If you use the :ref:`create <dtl-transforms>` and :ref:`subset <dataset_source>` properties to extract entities from a dataset and you do a full sync for deletion tracking, then subset in the source will still create entities that are not in the latest versions of that subset. 
+
+Therefore, you should materialise the subset in a separate dataset before “picking” your derived entities using subset to minimise the risk of creating entities that are not in the latest versions of your subset.
```
Comment (BaardBouvet) [inline patterns.rst:108]: When thinking about it, why would anyone then use subset to extract entities? The pattern we should describe is "How to extract entities", which is about using create and full scan to remove orphans. We can mention that subsets can't be used for this, and the solution is to just use filtering instead.

**Review by BaardBouvet — CHANGES_REQUESTED** [review]

File: patterns.rst (hunk 4 above)
Comment (JonasAls89) [reply patterns.rst]: Implemented

File: patterns.rst [hunk 5]
```diff
@@ -101,6 +101,12 @@ Optimistic locking
 ------------------
 Should be added via an external transform and then two hash values should be compared. In case of difference, discard entity.
 
+How to extract entities
+-----------------------
+In general, using :ref:`subset <dataset_source>` to extract entities is straight forward. Albeit, edge cases do exist. In example, if you use the :ref:`create <dtl-transforms>` and :ref:`subset <dataset_source>` properties to extract entities from a dataset following a full scan for deletion tracking, then subset in the source will still create entities that are not in the latest versions of that subset.
```
Comment (BaardBouvet) [inline patterns.rst:106]: The only way to extract entities is to use create, so this still needs a rewrite I think.

**Review by BaardBouvet — CHANGES_REQUESTED** [review]

File: patterns.rst (hunk 5 above)
Comment (JonasAls89) [reply patterns.rst]: Changed it.

**Review by BaardBouvet — APPROVED** [review]

**Pull request merged**